packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -0,0 +1,153 @@
+
+    try:
+        return template.render(**template_args)
+    except Exception:
+        raise RuntimeError(f'An error occurred while rendering template "{template_name}".')
+
+
+def _create_file(
+    template_name: str, file_name: str, directory_out: pathlib.Path, **template_args: Any
+) -> None:
+    output_file = directory_out / file_name
+
+    output = _render_template(template_name, **template_args).decode("utf-8")
+    formatted_output = black.format_str(
+        src_contents=output,
+        mode=black.Mode(line_length=100),
+    )
+    formatted_output = formatted_output.replace("\n", os.linesep)
+
+    with output_file.open("wb") as file:
+        file.write(formatted_output.encode("utf-8"))
+
+
+@click.command()
+@click.argument("module_name", type=str, default="")
+@click.option(
+    "-s",
+    "--measurement-service-class",
+    help="The service class of your measurement.",
+)
+@click.option(
+    "-o",
+    "--directory-out",
+    help="Output directory for Measurement Plug-In Client files. Default: '<current_directory>/<module_name>'",
+)
+def create_client(
+    module_name: str,
+    measurement_service_class: str,
+    directory_out: Optional[str],
+) -> None:
+    """Generates a Python Measurement Plug-In Client module for the measurement service.
+
+    You can use the generated module to interact with the corresponding measurement service.
+
+    MODULE_NAME: Name for the Python Measurement Plug-In Client module to be generated.
+    """
+    channel_pool = GrpcChannelPool()
+    discovery_client = DiscoveryClient(grpc_channel_pool=channel_pool)
+    import_modules: Dict[str, ImportType] = {}
+
+    try:
+        measurement_service_stub = get_measurement_service_stub(
+            discovery_client, channel_pool, measurement_service_class
+        )
+        registered_measurement_services = discovery_client.enumerate_services(
+            V2_MEASUREMENT_SERVICE_INTERFACE
+        )
+
+        if not registered_measurement_services:
+            click.ClickException(
+                "No active measurements were found. Please start one and try again."
+            )
+
+        selected_measurement_service = next(
+            (
+                measurement_service
+                for measurement_service in registered_measurement_services
+                if measurement_service.service_class == measurement_service_class
+            ),
+            None,
+        )
+
+        if selected_measurement_service is None:
+            click.ClickException(
+                f"Could not find any registered measurement with service class: '{measurement_service_class}'."
+            )
+            return
+
+        if not module_name.isidentifier():
+            module_name = get_python_module_name(
+                selected_measurement_service.service_class + "_client"
+            )
+        class_name = module_name.title().replace("_", "")
```

Comment:
I don't think deriving the class name from the module name works very well. The module name has a lot less information than the original service class, which can have dots for hierarchy and PascalCasing. The module name may have underscores, or it may not--many Python modules cramwordstogether.

Here's what I get from `poetry run ni-measurement-plugin-client-generator -s ni.examples.GameOfLife` (internal C# service):

![image](https://github.com/user-attachments/assets/3898c56e-d66c-43bf-bff1-c4e80ca93f53)

The service class started out as "ni.examples.GameOfLife". The generator set module_name to "niexamplesgameoflife_client". Then it converted the module_name to title case ("Niexamplesgameoflife_Client") and removed underscores, resulting in `NiexamplesgameoflifeClient`. 

If I use `poetry run ni-measurement-plugin-client-generator -s game_of_life_Python` the results are a little better because this example uses snake_case for its service class. 

![image](https://github.com/user-attachments/assets/16e2f59f-3bc4-47d9-b152-5643b62caaf8)

@dixonjoel That's weird, why did you make this example use a different service class naming scheme from everything else? 
![image](https://github.com/user-attachments/assets/e953df04-678a-4871-bce9-c8693a664432)

Anyway @Jotheeswaran-Nandagopal , please make the following changes to the class name handling:
- Add an option to set the class name independently of the module name.
- Base the default class name on the service class, not on the module name. 
- Make sure the default is reasonable for the examples (e.g. ni.examples.NIScopeAcquireWaveform_Python)

Reply:
Sure, I have set an option for `class-name` to get input from the user. If the user didn't provide any, default name will be taken from service class (capitalizing each word and omiting the invalid characters).

Reply:
Follow-up is in this thread: https://github.com/ni/measurement-plugin-python/pull/841#discussion_r1725950004

Reply:
I also filed https://github.com/ni/measurement-plugin-python/issues/849 about the game_of_life serviceClass.